server/routerlicious/packages/services-telemetry/src/resources.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Lumber } from "./lumber";
+
+export enum LogLevel {
+    Error,
+    Warning,
+    Info,
+    Verbose,
+    Debug,
+}
+
+export enum LumberType {
+    Metric,
+    Log,
+}
+
+export interface ILumberjackEngine {
+    emit(lumber: Lumber<string>): void;
+}
+
+export interface ITelemetryMetadata {
+    documentId: string | undefined;
```

Comment:
nit: use `?` syntax here instead of `| undefined`? For example
```ts
export interface ITelemetryMetadata {
  documentId?: string;
}
```

Reply:
If I do:
```
export interface ITelemetryMetadata {
    documentId?: string;
    tenantId?: string;
    clientId?: string;
    clientSequenceNumber?: number;
    sequenceNumber?: number;
}
```
wouldn't that allow somebody to define an object of type ITelemetryMetadata as
```
const x: ITelemetryMetadata = {
    documentId: "A",  
};
```
omitting the other properties? The idea @GaryWilber and I discussed was to make properties `string | undefined` explicitly so when they are not available, the user needs to provide "undefined". That's because we wanted to encourage the user to provide all data that is available for a given component... in other words, we want to avoid that if "clientId" is available, the user only provides documentId and tenantId. But if there is a better way to do that - or if that is something we don't need to enforce, I can definitely update how we define `ITelemetryMetadata`

Reply:
Ooh that is clever! I misunderstood when I was reading it, because functionally
```ts
const x: ITelemetryMetadata = { documentId: "A", };
```
is the same as
```ts
const x: ITelemetryMetadata = { documentId: "A",  tenantId: undefined, ... };
```
but I didn't think of it from the perspective of encouraging people to input all information possible! Perhaps drop a comment above the interface to explain that so it doesn't get lost or refactored away in the future?

Reply:
I am adding a comment for that, good point.